server/controllers/chartsController/helpers.js

```diff
@@ -251,41 +259,61 @@ export const postProcessData = (data, studyTotals) => {
   return processedDataBySite
 }
 
-export const mongoQueryFromFilters = (filters) => {
+export const mongoQueriesFromFilters = (filters, userAccess) => {
   if (!filters) {
     return
   }
   const activeFilters = []
-  const facet = {}
+  const includedCriteriaFacet = {}
   const chrCritFilters = filters.chrcrit_part
     .filter((f) => f.value === TRUE_STRING)
     .map((filter) => FILTER_TO_MONGO_VALUE_MAP[filter.name])
   const includedExcludedFilters = filters.included_excluded
     .filter((f) => f.value === TRUE_STRING)
     .map((filter) => FILTER_TO_MONGO_VALUE_MAP[filter.name])
+  const sexAtBirthFilters = filters.sex_at_birth
+    .filter((f) => f.value === TRUE_STRING)
+    .map((filter) => FILTER_TO_MONGO_VALUE_MAP[filter.name])
 
   if (!!chrCritFilters.length) {
-    facet.chrcrit_part = []
-    facet.chrcrit_part.push({
+    includedCriteriaFacet.chrcrit_part = []
+    includedCriteriaFacet.chrcrit_part.push({
       $match: { chrcrit_part: { $in: chrCritFilters } },
     })
```

Comment:
do we need to initialize the array and then push or can we just create the array with the match object directly?

```suggestion
    includedCriteriaFacet.chrcrit_part = [{
      $match: {
        chrcrit_part: { $in: chrCritFilters },
      },
    }]
```

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/c30a7d514c931ad36f83916c440ff7812449f315

Reply:
yes we have to push it, an empty object for a mongodb $facet leads to an error

Reply:
but it's not empty, it has an object in it 🤔

Reply:
ah good catch, thanks. https://github.com/AMP-SCZ/dpdash/pull/280/commits/4037a1df256411ef47d10350c00d735456ae08e9